core/src/main/resources/org/apache/spark/ui/static/executorspage.js

```diff
@@ -143,23 +143,205 @@ var sumOptionalColumns = [3, 4];
 var execOptionalColumns = [5, 6, 7, 8, 9, 10, 13, 14, 25];
 var execDataTable;
 var sumDataTable;
+var executorSummaryMetricsTableArray = [];
+var executorSummaryMetricsTableCurrentStateArray = [];
+var executorSummaryMetricsDataTable;
 
 function reselectCheckboxesBasedOnTaskTableState() {
   var allChecked = true;
+  var executorSummaryMetricsTableCurrentFilteredArray = executorSummaryMetricsTableCurrentStateArray.slice();
   if (typeof execDataTable !== "undefined") {
     for (var k = 0; k < execOptionalColumns.length; k++) {
       if (execDataTable.column(execOptionalColumns[k]).visible()) {
         $("[data-exec-col-idx=" + execOptionalColumns[k] + "]").prop("checked", true);
+        var executorSumCheckBoxId = $("[data-exec-col-idx=" + execOptionalColumns[k] + "]").attr('exec-sum-idx');
+        var selectedExecutorSummaryMetrics = executorSummaryMetricsTableArray.filter(row => (row.executorSumCheckBoxId).toString() == executorSumCheckBoxId)
+        for(var value in selectedExecutorSummaryMetrics) {
+          executorSummaryMetricsTableCurrentStateArray.push(selectedExecutorSummaryMetrics[value]);
+        }
+        executorSummaryMetricsTableCurrentFilteredArray = executorSummaryMetricsTableCurrentStateArray.slice();
       } else {
         allChecked = false;
       }
     }
   }
+
+  createDataTableForExecutorSummaryMetricsTable(executorSummaryMetricsTableCurrentFilteredArray)
+
   if (allChecked) {
     $("#select-all-box").prop("checked", true);
   }
 }
 
+function getColumnNameForExecutorMetricSummary(columnKey) {
+  switch(columnKey) {
```

Comment:
Shall we use a string map instead of switch?

Reply:
> Shall we use a string map instead of switch?

Hmm, same method from stagingspage.js

Reply:
> Shall we use a string map instead of switch?

Updated